<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [2]:
# Membuat dataframes dari dictionary
import pandas as pd

data = {'nomor': [3, 2, 1, 0], 'huruf': ['a', 'b', 'c', 'd']}
pd.DataFrame.from_dict(data)

nomor huruf
0      3     a
1      2     b
2      1     c
3      0     d

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [3]:
# Membaca data dan memasukannya kedalam suatu variabel
df_from_csv = pd.read_csv('diamonds.csv')
print(df_from_csv)
# Menampilkan first and last data
#top = df_from_csv.head(5)
#bottom = df_from_csv.tail(5)
#concatenated = pd.concat([top,bottom])
#concatenated

       Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

<h4>-Apa perbedaan Series dengan list dan dictionary</h4>

Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data seperti integer, string, dan lain sebagainya. tipe data dari objek series ini harus seragam. 

List dan Dictionary

List adalah salah satu tipe data untuk mengoleksikan data di python.

Dictionary adalah salah satu jenis tipe data di python yang memetakan antara key dan value dari suatu data.

<h4>- Apa perbedaan dataframe dan series</h4>

DataFrame adalah suatu objek 2 dimensi tempat menyimpan data dengan lebih terstruktur. dataframe memiliki 2 index, yaitu index baris dan index columns. Dalam satu column dataframe harus memiliki tipe data yang sama. tapi antar columnnya dataframe bisa memiliki jenis data yang berbeda. Sedangkan series hanya memiliki satu sumbu saja.

In [6]:
# Contoh data Series

x = pd.Series(["a","b","c","d"])
print(x)

0    a
1    b
2    c
3    d
dtype: object


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [12]:
# code here
top1 = df_from_csv['price'].head(1)
bottom1 = df_from_csv['color'].tail(1)
concatenated1 = pd.concat([top1,bottom1])
concatenated1

0        326
53939      D
dtype: object

<h4>- baris pertama dari column price  : 326</h4>
<h4>- baris terakhir dari column color : D</h4>

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab:  Statistik deskriptif adalah salah satu bagian dari ilmu statistika yang berhubungan dengan aktivitas penghimpunan, penataan, peringkasan dan penyajian data dengan harapan agar data lebih bermakna, mudah dibaca dan mudah dipahami oleh pengguna data. Statistik deskriptif hanya sebatas memberikan deskripsi atau gambaran umum tentang karakteristik objek yang diteliti tanpa maksud untuk melakukan generalisasi sampel terhadap populasi.

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [18]:
# Gunakan suatu metode dari pandas 
#untuk mengetahui deskriptif statistik dari suatu data
df_from_csv.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

In [20]:
#Berapa nilai rata-rata dari column price
df_from_csv['price'].mean()

3932.799721913237

In [21]:
#Berapa nilai standar deviasi dari column depth
df_from_csv['depth'].std()

1.4326213188337733

In [22]:
#Berapa nilai maximum dari column carat
df_from_csv['carat'].max()

5.01

Jawab: